connect/src/com/telenor/connect/AbstractSdkProfile.java

```diff
@@ -166,6 +166,10 @@ protected void previewParameters(ParametersHolder parameters) {
                         && isEmpty(wellKnownConfig.getNetworkAuthenticationTargetUrls()))) {
                     parameters.add("login_hint", String.format("MCCMNC:%s", mccMnc));
         }
+
+        if (!ConnectSdk.isCellularDataNetworkConnected()) {
```

Comment:
The bug basically lies in the logical decomposition of the problem. The "opposite" of cellular connection is not WiFi, but a non-cellular connection. Besides, even with WiFi in picture, take a look at the following scenario: 
If you
1. start the application
2. turn Flight mode on
3. turn WiFi on
4. click on the CONNECT button
your patch will force HE over a disconnected cellular network. To tell the whole truth, you do catch IOException to mitigate this problem, but still, I think the code would be more efficient if you refrained from forcing anything through a network that you know is disconnected.

To illustrate my point, take a look at the patch where I tried to fix the problem: https://github.com/telenordigital/connect-android-sdk/pull/67 .